swayfmt/CONTRIBUTING.md

```diff
@@ -0,0 +1,81 @@
+# Contributing to the Sway Language Formatter
+
+Firstly, thank you for taking interest in advancing the Sway language formatter! This guide will walk you through installation and best practices for contributing to this project.
+
+## Installation
+
+> **Pre-Installation:** If you've previously installed `forc-fmt` via `fuelup`, you will need to remove its binary in order to use the binary compiled from source.
+
+```sh
+# find fuelup `forc-fmt` binary
+where forc-fmt
+# output: `~/.fuelup/bin/forc-fmt`
+#
+# remove fuelup `forc-fmt` binary
+rm ~/.fuelup/bin/forc-fmt
+```
+
+In order to see `swayfmt` in action you will need both `swayfmt` and the [`forc-fmt`](../forc-plugins/forc-fmt/) plugin. As previously stated, you can install these via [`fuelup`](https://github.com/FuelLabs/fuelup) which will install the complete Fuel toolchain, or follow the instructions below to install from source.
+
+> **Note:** `cargo` is a prerequisite to this build.
+
+```sh
+# 1. move to your preferred directory
+#    example: cd ~/Code/
+#
+# 2. clone the Sway repo
+git clone https://github.com/FuelLabs/sway.git
+#
+# 3. build from manifest and move the compiled result to your `.cargo/bin` folder
+cargo build --manifest-path ~/sway/forc-plugins/forc-fmt/Cargo.toml && mv ~/sway/target/debug/forc-fmt ~/.cargo/bin
+```
+
+### Testing
+
+> **Testing via Fuelup:**
+
+Move to a folder of your project containing a `Forc.toml`, and run the command:
+
+```sh
+forc fmt
+```
+
+> **Testing from source:**
+
+Move to a folder of your project containing a `Forc.toml`, and execute the binary:
+
+```sh
+forc-fmt
+```
+
+Be sure that your code compiles before executing the command!
+
+### Contribution Guidelines
+
+**Issues:**
+
+- Please check existing issues before opening a new ticket.
+- If there is a bug that isn't represented by a ticket please provide the following details:
+  - `forc` version (run the command `forc --version`)
+  - `rust` version (run the command `rustc --version`)
+  - operating system and version
+  - copy and paste the following into separate and labeled code blocks:
+    - error output, if any
+    - the full code snippet the problem originated from
+    - the current output after formatting
+    - the desired output after formatting
```

Comment:
we could just use an [issue template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository) for this

Reply:
#2725